clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/HonoKafkaConsumer.java

```diff
@@ -1260,4 +1306,48 @@ private ExecutorService getKafkaConsumerWorker(final KafkaConsumer<String, V> co
         }
         return worker;
     }
+
+    private final class SubscriptionUpdateTracker {
+        private final Promise<Void> outcome = Promise.promise();
+        private final String topicName;
+        private final AtomicInteger rebalancesLeft = new AtomicInteger(10);
+
+        /**
+         * Creates a new tracker for a topic name.
+         */
+        SubscriptionUpdateTracker(final String topicName) {
+            this.topicName = Objects.requireNonNull(topicName);
+        }
+
+        Future<Void> outcome() {
+            return this.outcome.future();
+        }
+
+        String getTopicName() {
+            return topicName;
+        }
+
+        boolean isContainedInSubscribedTopics(final Set<String> topics) {
+            rebalancesLeft.decrementAndGet();
```

Comment:
Doing `rebalancesLeft.decrementAndGet(` in this method feels rather unexpected.
A separate method for this would make things clearer, FMPOV.
(Returning the new `rebalancesLeft` value there could make the `hasRebalancesLeft()` method unnecessary.)

Reply:
But then the calling code needs to keep track of the counter and that is what I wanted to encapsulate. So, I would rather keep it the way it is now.

Reply:
How about renaming the method to `isContainedInSubscribedTopicsAfterRebalance()` then, to give a hint about the side-effect of the method?